src/main/java/org/breskul/pool/PooledDataSource.java

```diff
@@ -131,4 +131,9 @@ private static PooledDataSource createDataSourceInstanceFromProperties() {
         return new PooledDataSource(url, username, password, poolSize);
     }
 
+//    Method for unit tests purpose only!
```

Comment:
You can use @VisibleForTesting annotation. See line 73 in this class for instance

Reply:
We can't because test where we execute are not at the same package. This is valid idea, but I don't want to have a bunch of tests distributed in multiple folders. But this is good point to discuss issue with testing Singleton. Also I am nit sure that it make sense to create Class as singleton ...